### PR TITLE
Update document on load testing and scripts

### LINKED ISSUE #1153: [DOCS] Update benchmark.md
## 📚 Documentation

The benchmark.md should be the go-to reference document on how to load test and measure the engine's performance,
but currently it's outdated, we should update it with the following guidelines:
1. First of all, ALWAYS register the specs and OS of the machines you’re using.
2. Write down every relevant command/function that you use, be it the shell or the codebase of the game.
3. Also, always write down each parameter combination you’re using.
4. Use multiple sources for load testing and benchmarks, like htop, etop, fprop, NewRelic.
5. Pictures/ Logs > Videos, maybe use videos when performance affects UX.
6. Log your results on a folder on the repo, this is important since we want to track advancements and regressions.
7. Of course, update the API of the functions accordingly.
8. This kind of benchmark is experimental, but we should always do the same
to have a valid way to compare if we’ve advanced or regressed.

### PR DESCRIPTION
**Pull request by mFragaBA**
Closes #1153

## Motivation
With the load testing changes from #1132, we need to update the current documentation. Both as a means of having instructions to setup the servers to run the load tests and to give guidelines on how to run them and write reports so anyone can run load tests without greater inconveniences.

## Summary of changes

- updated benchmarking documentation on `docs/src/benchmarking.md`
- created a general load testing documentation on `docs/src/load_testing.md`
- added scripts on `server/load_test` to setup both the server that runs the game and the one that runs the load test

## How has this been tested?

Went through step by step of the instructions on `load_testing.md` from a clean server and got the load test info.

## Checklist
- [X] I have tested the changes locally.
- [X] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [X] This change requires new documentation.
  - [X] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

File: docs/src/load_testing.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+The goal of load testing is to simulate a real scenario where there
+are a lot of players in one server, our current goal is 1000 players by server.
+
+Our current methodology is to use 2 servers to load test:
+- The game server: where the game's server is actually running.
+- The load test client: sends requests to the game server, we 
+  want a dedicated load test client to avoid consumer network
+  and hardware limitations.
+
+Soem things to keep in mind about load tests:
```
Comment (mFragaBA) [inline docs/src/load_testing.md:11]:
```suggestion
Some things to keep in mind about load tests:
```

File: docs/src/load_testing.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- Always write a report, and every report must go to:
+  server/load_test/reports/your_report.md, this is important 
+  to keep track of possible performance regressions.
+- Always write down the specs + OS + config from where you're running the tests.
+  For example, did you change linux's governor settings? Write it down.
+  Did you test any BEAM VM flag? Keep a record of it.
+- Write down every snippet code that you've used, be it Elixir, Rust or bash commands,
```
Comment (mFragaBA) [inline docs/src/load_testing.md:18]:
```suggestion
- Write down every code snippet that you've used, be it Elixir, Rust or bash commands,
```

File: docs/src/load_testing.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  if you think it's obvious, write it down just in case.
+- Track records of the parameters you've used for the tests, eg: Player Amount,
+  where there bots enabled? etc.
```
Comment (mFragaBA) [inline docs/src/load_testing.md:21]:
```suggestion
  were there bots enabled? etc.
```

File: docs/src/load_testing.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+- Use text or pictures for reports, videos only if there are UX (i.e. gameplay) issues.
+- Use multiple sources of truth and data, like htop, New Relic, erlang's etop/fprop.
+- Feel free to experiment a bit, certain VM flags can improve or hinder performance,
```
Comment (mFragaBA) [inline docs/src/load_testing.md:24]: perhaps we can also ask for a reason for a flag to be used

File: server/load_test/setup_load_client.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
+export MIX_ENV=prod
+cd /tmp
+# Only clone `server/` subdirectory
```
Comment (mFragaBA) [inline server/load_test/setup_load_client.sh:3]: this doesn't apply anymore

**Comment by mFragaBA** [comment]
we should add a reference to `server/load_test/reports/some_report_file.md` to have as an example and be used as guide for further reports

**Review by mFragaBA — COMMENTED** [review]
we should remove/update `benchmarking.md`. I'm more inclined to removing it and moving relevant info to `load_testing.md`

File: docs/src/load_testing.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+   ```sh
+   chmod +x ./setup_game_server.sh && ./setup_game_server <BRANCH_NAME_TO_TEST>
+   ```
+
+4. Now you can start the game server with: 
+```sh
+   systemctl daemon-reload && systemctl start curse_of_myrra
+```
+   You can check the logs with `journalctl -xefu curse_of_myrra`.
+5. Make sure to disable hyperthreading, if using an x86 CPU:
```
Comment (marco-paulucci) [inline docs/src/load_testing.md:60]: Shouldn't this be step 4?

Comment (mFragaBA) [reply docs/src/load_testing.md]: yes, it should!

File: docs/src/load_testing.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+   ```
+
+   `setup_game_server` can also take a branch name as an argument. So if you want to run the load test on an specific branch, you can instead do:
+   ```sh
+   chmod +x ./setup_game_server.sh && ./setup_game_server <BRANCH_NAME_TO_TEST>
+   ```
+
+4. Now you can start the game server with: 
+```sh
+   systemctl daemon-reload && systemctl start curse_of_myrra
```
[the remark this answers is not in the document]
Comment (mFragaBA) [reply docs/src/load_testing.md]: fixed

File: docs/src/load_testing.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+2. If it's not already there, copy the script on this repo under
+   `server/load_test/setup_game_server.sh` it clones the game server, compiles it, and creates a
+   systemd service for it, run it with:
+   ```sh
+   chmod +x ./setup_game_server.sh && ./setup_game_server
+   ```
+
+   `setup_game_server` can also take a branch name as an argument. So if you want to run the load test on an specific branch, you can instead do:
+   ```sh
+   chmod +x ./setup_game_server.sh && ./setup_game_server <BRANCH_NAME_TO_TEST>
```
[the remark this answers is not in the document]
Comment (mFragaBA) [reply docs/src/load_testing.md]: fixed

File: docs/src/load_testing.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+### Game Server Setup
+1. Log into it with ssh: 
+   ```sh
+   ssh myuser@myrra_load_test_server
+   ```
+2. If it's not already there, copy the script on this repo under
+   `server/load_test/setup_game_server.sh` it clones the game server, compiles it, and creates a
+   systemd service for it, run it with:
+   ```sh
+   chmod +x ./setup_game_server.sh && ./setup_game_server
```
[the remark this answers is not in the document]
Comment (mFragaBA) [reply docs/src/load_testing.md]: fixed

**Comment by mFragaBA** [comment]
closing this as it will be moved with the rest of the backend to the [game backend repo](https://github.com/lambdaclass/game_backend)

**Pull request closed without merging**